src/mapping/query/dynamic_regex.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{event::Value, mapping::Result};
+use bytes::Bytes;
+use std::{collections::BTreeMap, convert::TryFrom};
+
+#[derive(Debug, Clone)]
+pub struct DynamicRegex {
+    pattern: String,
+    multiline: bool,
+    insensitive: bool,
+    global: bool,
+    compiled: Option<Result<regex::Regex>>,
```

Comment:
I would assume whatever function compiles the regex, return an error, so that we only store valid regex's here?

Reply:
I left the `Result` in so if it has an invalid regex the error is stored here. Otherwise on an invalid regex it would just compile it every time it is accessed.